Lib/asyncio/unix_events.py

```diff
@@ -307,6 +307,112 @@ def create_unix_server(self, protocol_factory, path=None, *,
         return server
 
+    @coroutine
+    def create_unix_datagram_connection(self, protocol_factory, path=None, *,
+                                        sock=None):
+        if path is not None:
+            if sock is not None:
+                raise ValueError(
+                    'path and sock can not be specified at the same time')
+
+            path = _fspath(path)
+            sock = socket.socket(socket.AF_UNIX, socket.SOCK_DGRAM)
+            try:
+                sock.setblocking(False)
+                yield from self.sock_connect(sock, path)
+            except:
+                sock.close()
+                raise
+        else:
+            if sock is None:
+                raise ValueError(
+                    'path was not specified, and no sock specified')
+
+            if (sock.family != socket.AF_UNIX or
+                    not base_events._is_dgram_socket(sock)):
+                raise ValueError(
+                    'A UNIX Domain Datagram Socket was expected, got {!r}'
+                    .format(sock))
+
+            sock.setblocking(False)
+            path = sock.getpeername()
+
+        protocol = protocol_factory()
+        waiter = self.create_future()
+        transport = self._make_socket_datagram_transport(sock, protocol, path,
+                                                         waiter)
+
+        try:
+            yield from waiter
+        except:
+            transport.close()
+            raise
+
+        return transport, protocol
+
+    @coroutine
+    def create_unix_datagram_server(self, protocol_factory, path=None, *,
+                                    sock=None):
+        if path is not None:
+            if sock is not None:
+                raise ValueError(
+                    'path and sock can not be specified at the same time')
+
+            path = _fspath(path)
+            # Check for abstract socket. `str` and `bytes` paths are supported.
+            if path[0] not in (0, '\x00'):
+                try:
+                    if stat.S_ISSOCK(os.stat(path).st_mode):
+                        os.remove(path)
+                except FileNotFoundError:
+                    pass
+                except OSError as err:
+                    # Directory may have permissions only to create socket.
+                    logger.error('Unable to check or remove stale UNIX'
+                                 ' socket %r: %r', path, err)
+
+            sock = socket.socket(socket.AF_UNIX, socket.SOCK_DGRAM)
+            try:
+                sock.setblocking(False)
+                sock.bind(path)
+            except OSError as exc:
+                sock.close()
+                if exc.errno == errno.EADDRINUSE:
+                    # Let's improve the error message by adding
+                    # with what exact address it occurs.
+                    msg = 'Address {!r} is already in use'.format(path)
+                    raise OSError(errno.EADDRINUSE, msg) from None
```

Comment:
This was made to improve the error message by including the address that failed the bind. I'm not sure chaining the exception will provide more information on the error.